File: ospf-kotlin-example/src/main/fuookami/ospf/kotlin/example/framework_demo/demo1/domain/route_context/model/Service.kt
package fuookami.ospf.kotlin.example.framework_demo.demo1.domain.route_context.model

import fuookami.ospf.kotlin.utils.concept.*
import fuookami.ospf.kotlin.utils.math.UInt64

class Service(
    val id: UInt64,
    val capacity: UInt64,
    val cost: UInt64
) : AutoIndexed(Service::class) {
    override fun toString() = "S$id"
}

File: ospf-rust-example/src/framework/demo1/route_context/model/service.rs
//! 服务模型 / Service model

/// 服务模型 / Service model (对齐 Kotlin Service)
#[derive(Debug, Clone)]
pub struct Service {
    /// 服务标识 / Service identifier
    pub id: u64,
    /// 服务容量 / Service capacity
    pub capacity: f64,
    /// 服务成本 / Service cost
    pub cost: f64,
}
